src/internal/data/data_collection.go

```diff
@@ -70,6 +70,12 @@ type Stream interface {
 	Deleted() bool
 }
 
+// Locationer provides a LocationPath describing the path with Display Names
+// instead of canonical IDs
+type LocationPather interface {
+	LocationPath() path.Path
```

Comment:
This is something we haven't fully discussed, but are we comfortable with always returning a Path that has all the leading elements like tenant, service, user, and category?

I guess if we intend to use this for selectors it's kind of required right? After all, Path is meant to help with selector comparisons by providing index-independent access to fields like `Category()` etc

Reply:
I see we're actually storing a subset of this information. Future PRs can revisit if we need to return an actual Path from this or if something less strict like a string would be alright

Reply:
I think it's most holistic for us to maintain the full path here.  Passing around paths that arbitrarily do or do not comply with the standard format is bound produce later gotchas and headaches, not to mention increasing cognitive load to track which is which.